server/src/main/java/org/opensearch/common/blobstore/transfer/UploadFinalizer.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.transfer;
+
+/**
+ * UploadFinalizer is an interface with support for a method that will be called once upload is complete
+ *
+ * @opensearch.internal
+ */
+public interface UploadFinalizer {
+    void accept(boolean uploadSuccess);
+}
```

Comment:
Can we not reuse `BooleanConsumer ` here?

Reply:
New interface is just to provider a clear purpose of the interface since it is shared with plugin. We can replace it with internal FI as well. You decide.

Reply:
Using `Consumer<Boolean>` and removed `UploadFinalizer`.